mpi-proxy-split/mpi-wrappers/mpi_p2p_wrappers.cpp

```diff
@@ -256,16 +256,35 @@ USER_DEFINED_WRAPPER(int, Sendrecv_replace, (void *) buf, (int) count,
                      (int) sendtag, (int) source,
                      (int) recvtag, (MPI_Comm) comm, (MPI_Status *) status)
 {
-
-  // Send first
   int retval;
-  retval = MPI_Send(buf, count, datatype, dest, sendtag, comm);
-  if (retval != MPI_SUCCESS) {
-    return retval;
+
+  // Send/recv must occur in different order to avoid deadlock
+  if (source > dest) { // Send first, then receive
```

Comment:
Wouldn't the source and destination ranks for every rank participating in the MPI_Sendrecv_replace be same?

Reply:
I meant, for rank 0 if send = 0 , dest = 1, then for rank 1 also it will be send = 0, dest = 1.

Reply:
checking

Reply:
Realized that source/dest aren't necessarily the same, fixed now to be independent of that (no deadlock either way)